shop/shop-domain/src/main/java/shop/woowasap/shop/app/spi/ProductRepository.java

```diff
@@ -10,6 +10,8 @@ public interface ProductRepository {
 
     Optional<Product> findById(long productId);
 
+    Optional<Product> findByIdAndValidSaleTime(long productId);
+
     ProductsPaginationResponse findAllValidWithPagination(int page, int size);
```

Comment:
오잉? 여기는 final이 없는거 같네요 확인 한번만 부탁드립니다~

```suggestion
    Optional<Product> findById(final long productId);

    Optional<Product> findByIdAndValidSaleTime(final long productId);

    ProductsPaginationResponse findAllValidWithPagination(final int page, final int size);
```

Reply:
추가했습니당!